src/PaymentGateways/ServiceProvider.php

```diff
@@ -34,5 +35,15 @@ public function boot()
         Hooks::addFilter('give_register_gateway', RegisterPaymentGateways::class);
         Hooks::addFilter('give_payment_gateways', RegisterPaymentGatewaySettingsList::class);
         Hooks::addAction('template_redirect', GatewayRoute::class);
+
+        // We are adding event listener to this action hook.
+        // This hook fires for PayPal Standard ipn for onetime donation.
+        Hooks::addAction(
```

Comment:
Does a gateway route method not work in this case? If possible, I would suggest leveraging gateway route methods for IPN handlers instead of adding a hook outside of the context of the gateway itself.

Reply:
@kjohnson  this was an old way to handle IPN event. I updated logic: https://github.com/impress-org/givewp/compare/55949ae3613b...d41881582b52